src/libdrakvuf/libdrakvuf.h

```diff
@@ -274,6 +302,32 @@ char *drakvuf_get_current_process_name(drakvuf_t drakvuf,
                                        uint64_t vcpu_id,
                                        x86_registers_t *regs);
 
+
+bool drakvuf_get_current_thread_id( drakvuf_t drakvuf, 
+                                    uint64_t vcpu_id, 
+                                    x86_registers_t *regs,
+                                    uint32_t *thread_id );
+
+bool drakvuf_get_previous_mode( drakvuf_t drakvuf, 
```

Comment:
This function should have a comment at least its intended use-case. I think it may be more clear if it was named drakvuf_get_thread_previous_mode. BTW, when is the field in the kthread struct get updated by Windows? In other words, what does this mode field actually signify? I assume the CPL of the thread when it was scheduled out (ie, a trapframe is pushed on its stack), but in this case this is the active thread, so isn't the value stale right now?